libvast/src/type.cpp

```diff
@@ -462,6 +462,47 @@ caf::optional<offset> record_type::offset_from_index(size_t i) const {
   return it->offset;
 }
 
+caf::expected<record_type>
+merge(const record_type& lhs, const record_type& rhs) {
+  record_type result = lhs;
+  std::map<std::string, type> all_fields;
```

Comment:
Using a map makes little sense here, use a flat scan instead.
Write a story to use a `stable_map` for the fields of a record type.